rllib/evaluation/env_runner_v2.py

```diff
@@ -530,6 +516,10 @@ def _process_observations(
                 continue
 
             episode: EpisodeV2 = self._active_episodes[env_id]
+            # If this episode is brand-new, call the episode start callback(s).
+            # Note: EpisodeV2s are initialized with length=-1 (before the reset).
+            if episode.length == -1:
```

Comment:
can I suggest we don't rely on the internals of EpisodeV2 directly?
it will be better if we create an API on EpisodeV2 similar to Episode
```
@property
def started(self) -> bool:
    return bool(self._has_init_obs)
```
then we can do:
```
if not episode.started:
    self._call_on_episode_start(episode, env_id)
```
it's likely safer if we rely on self._has_init_obs rather than the initial value of length?

Reply:
another quick question is why do we do this here.
why don't we self._call_on_episode_start(episode, env_id) right after the reset() op is done?
although we may need to do this at a couple of places, I feel like it's mentally easier if they happen one after another right away?

Reply:
We do `next_episode_length = episode.length + 1` right after this, though :) 
Also accessing the episode's internal properties.

Either way: Fixed it, added the suggested property.

Reply:
huh, remote env ... 😢 
thanks for the fix though. feels a bit better we don't rely on that -1.